packages/docprovider/src/yprovider.ts

```diff
@@ -70,69 +49,56 @@ export class WebSocketProvider
     user.changed.connect(userChanged);
   }
 
-  get renameAck(): Promise<boolean> {
-    return this._renameAck.promise;
-  }
-
-  setPath(newPath: string): void {
-    if (newPath !== this._path) {
-      this._path = newPath;
-      const encoder = encoding.createEncoder();
-      this._renameAck = new PromiseDelegate<boolean>();
-      encoding.write(encoder, 127);
-      // writing a utf8 string to the encoder
-      const escapedPath = unescape(
-        encodeURIComponent(
-          this._format + ':' + this._contentType + ':' + newPath
-        )
-      );
-      for (let i = 0; i < escapedPath.length; i++) {
-        encoding.write(
-          encoder,
-          /** @type {number} */ escapedPath.codePointAt(i)!
+  get ready(): Promise<boolean> {
+    const serverSettings = ServerConnection.makeSettings();
+    const url = URLExt.join(
+      serverSettings.baseUrl,
+      FILE_PATH_TO_ROOM_ID_URL,
+      encodeURIComponent(this._path)
+    );
+    const data = {
+      format: this._format,
+      type: this._contentType
+    };
+    const promise = ServerConnection.makeRequest(
+      url,
+      {
+        method: 'PUT',
+        body: JSON.stringify(data)
+      },
+      serverSettings
+    )
+      .then(response => {
+        if (response.status !== 200 && response.status !== 201) {
```

Comment:
We should make sure the server never gives back an `HTTP 204`, (no content) response, which is also valid. This is a minor point and only relevant if that's how the back-end is implemented.

Reply:
Currently (actually it's not merged yet) the [server doesn't send back `HTTP 204`](https://github.com/jupyter-server/jupyter_server_ydoc/pull/41/files#diff-6b5044ef7a07b2a53a9621cf35e5c3a5166e42b669be8b11b6d06ed9f33b1f5eR296-R324), but it doesn't hurt to add that condition maybe?

Reply:
Actually, no since the server doesn't send `HTTP 204`.